### PR TITLE
feat: add "_to_patch_data" to "DataFrame" to support updating data

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2h0za7f 实现支持更新数据的 Operation](https://app.clickup.com/t/2h0za7f)

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -303,6 +311,17 @@ def _extend(self, values: "DataFrame") -> None:
         for key, value in self._columns.items():
             value._extend(values[key])  # pylint: disable=protected-access
 
+    def _to_post_data(self) -> List[Dict[str, Any]]:
+        names = self._column_names.copy()
+        values = list(self._columns.values())
+
+        names.append(RECORD_KEY)
+        values.append(self._record_key.to_pyarrow())  # type: ignore[union-attr]
```
Comment (linjiX) [inline graviti/dataframe/frame.py:319]: shall we change the type of `_record_key` to ColumnSeries?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py (hunk 1 above)
Comment (graczhual) [reply graviti/dataframe/frame.py:319]: a special ColumnSeries without schema?

Comment (linjiX) [reply graviti/dataframe/frame.py:319]: _record_key is string, it can have schema pt.string()

File: graviti/dataframe/frame.py [hunk 2]
```diff
@@ -627,12 +649,13 @@ def extend(self, values: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
         elif self.operations is not None:
             values = values.copy()
 
+        values_length = len(values)
```
Comment (linjiX) [inline graviti/dataframe/frame.py:652]: why we need `values_length`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py (hunk 2 above)
Comment (graczhual) [reply graviti/dataframe/frame.py:652]: if we use extend like `df.extend(df)`, then the `len(values)` will double

**Review by linjiX — APPROVED** [review]

**Review by AChenQ — APPROVED** [review]

**Pull request merged**